### PR TITLE
Upgrading Plugin from Moodle 2.6 to Moodle 3.4+

### PR DESCRIPTION
**Pull request by andrewnagyeb**
Features
- **New!** : Compatability to Moodle 3.4+.
- **New!** : Ideone judge engine is upgraded to sphere-engine.
- **New!** : Syntax Highlighter is now back since moodle-onlinejudge 1.9.
- **New!** : Student statistics is shown again.

### CODE REVIEW

File: clients/mod/assign/feedback/onlinejudge/lang/pt_br/assignfeedback_onlinejudge.php [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+/**
+ * Strings for Online Judge Assignment Type
+ *
+ * @package   local_onlinejudge
+ * @copyright 2011 Sun Zhigang (http://sunner.cn)
+ * @author    Sun Zhigang
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+$string['testcasemanagement'] = '测试用例管理';
```
Comment (sunner) [inline clients/mod/assign/feedback/onlinejudge/lang/pt_br/assignfeedback_onlinejudge.php]: This is Chinese, not pt_br :-)

File: clients/mod/assign/feedback/onlinejudge/backup/moodle2/backup_assignment_onlinejudge_subplugin.class.php [hunk 2]
```diff
@@ -27,12 +28,14 @@
  * backup subplugin class that provides the necessary information
  * needed to backup one assignment->onlinejudge subplugin.
  */
-class backup_assignment_onlinejudge_subplugin extends backup_subplugin {
+class backup_assignfeedback_onlinejudge_subplugin extends backup_subplugin
+{
```
Comment (sunner) [inline clients/mod/assign/feedback/onlinejudge/backup/moodle2/backup_assignment_onlinejudge_subplugin.class.php]: [Moodle coding style](https://docs.moodle.org/dev/Coding_style) suggests no single line "{". We'd best to follow it.

File: clients/mod/assign/feedback/onlinejudge/lang/pt_br/assignfeedback_onlinejudge.php (hunk 1 above)
Comment (andrewnagyeb) [reply clients/mod/assign/feedback/onlinejudge/lang/pt_br/assignfeedback_onlinejudge.php]: I have fixed that, by removing it and left it as a todo until someone who can speak the language translate it.

File: clients/mod/assign/feedback/onlinejudge/backup/moodle2/backup_assignment_onlinejudge_subplugin.class.php (hunk 2 above)
Comment (andrewnagyeb) [reply clients/mod/assign/feedback/onlinejudge/backup/moodle2/backup_assignment_onlinejudge_subplugin.class.php]: Whole files are now updated with the above coding style.

File: clients/mod/assign/feedback/onlinejudge/lang/zh_cn/assignfeedback_onlinejudge.php [hunk 3]
```diff
@@ -148,4 +148,15 @@
 $string['typeonlinejudge'] = '在线评测';
 $string['usefile'] = '测试用例来自文件';
 $string['waitingforjudge'] = '等待评测结果';
+$string['enabled_help'] = 'Allowing submissions to be programmatically judged';
```
Comment (sunner) [inline clients/mod/assign/feedback/onlinejudge/lang/zh_cn/assignfeedback_onlinejudge.php]: Please remove the new English strings from zh-cn and pt-br lang file. Moodle will use English string directly if meet missing string in other languages.

Comment (andrewnagyeb) [reply clients/mod/assign/feedback/onlinejudge/lang/zh_cn/assignfeedback_onlinejudge.php]: Done.

**Pull request merged**